### PR TITLE
glance: minify glance config templates

### PR DESCRIPTION
**Pull request by Itxaka**

### CODE REVIEW

**Comment by Itxaka** [comment]
[(5) Update templates for config files for Newton - Glance](https://trello.com/c/vygtc8vY/373-5-update-templates-for-config-files-for-newton-glance)

**Comment by Itxaka** [comment]
@aplanas I thougth about that seeing your comments on the heat minimise PR, but I would really-really like to do this in 2 or more PRs. 1 to minimize them, another to move them to the proper `component.conf.d`, another one for upgrading them to newton levels+deprecated stuff

Do you agree with this or want me to just do one PR per component to minimise+move them to the proper dir?

**Comment by aplanas** [comment]
@Itxaka for me makes sense to separate the newton one, not sure about the first two. IMO those are the same and the run in mkcloud will test something of value: that the --config-dir is working, the directory exists in the package and the old configuration is still working.

Again IMO there is little value on testing the use of `sed` to remove the lines that starts with `#`, because the big change is in the conf.d thingy.

Basically having the CI in red will point a problem in the package, and this is something that we need to know ASAP

**Comment by Itxaka** [comment]
@aplanas sounds fair to me, updating both PRs!

**Comment by aplanas** [comment]
@Itxaka thanks you a lot! Now to fix the reds : ))

**Comment by Itxaka** [comment]
will squash later if needed, but this way its easier to check for differences without having to load the -4000 lines from the first commit

**Comment by aplanas** [comment]
@Itxaka can you fix the `cinder:` prefix from the commits? I guess that is `glance:`

**Comment by Itxaka** [comment]
@aplanas woops, fixed.

Failing anyway: 
```
00:49:44.014 Chef::Exceptions::EnclosingDirectoryDoesNotExist
00:49:44.014 ------------------------------------------------
00:49:44.014 Parent directory /etc/glance/glance.conf.d does not exist.
```

Package needs to be fixed as it creates `/etc/glance/conf.d/` dir. Other packages create `/etc/{component}/{component}.conf.d` instead (keystone, heat) so Im guessing thats the rigth one.

**Comment by aplanas** [comment]
@Itxaka there are SR for the packages already. The links are in the --config-dir card.

**Comment by toabctl** [comment]
retriggered gating...

**Comment by Itxaka** [comment]
glance api conf needs to go under /etc/glance/glance-api.conf.d/ as it shares keywords with the glance registry :o

**Comment by Itxaka** [comment]
rebased plus (hopefully) fixed

File: chef/cookbooks/glance/attributes/default.rb [hunk 1]
```diff
@@ -37,22 +35,22 @@
...
 default[:glance][:api][:bind_port] = "9292"
 default[:glance][:api][:log_file] = "/var/log/glance/api.log"
-default[:glance][:api][:config_file]="/etc/glance/glance-api.conf"
+default[:glance][:api][:config_file] = "/etc/glance/glance-api.conf.d/100-glance-api.conf"
 default[:glance][:api][:service_name] = "glance-api"
 
 default[:glance][:registry][:bind_port] = "9191"
 default[:glance][:registry][:log_file] = "/var/log/glance/registry.log"
-default[:glance][:registry][:config_file]="/etc/glance/glance-registry.conf"
+default[:glance][:registry][:config_file] = "/etc/glance/glance-registry.conf.d/100-glance-registry.conf"
```
Comment (nicolasbock) [inline chef/cookbooks/glance/attributes/default.rb]: You could try to put that whole string on a second line instead of splitting it. Maybe that would be enough to satisfy hound :)

Comment (Itxaka) [inline chef/cookbooks/glance/attributes/default.rb]: Good point, done!

File: chef/cookbooks/glance/attributes/default.rb [hunk 2]
```diff
@@ -37,22 +35,24 @@
...
 default[:glance][:api][:log_file] = "/var/log/glance/api.log"
-default[:glance][:api][:config_file]="/etc/glance/glance-api.conf"
+default[:glance][:api][:config_file] = "/etc/glance/glance-api.conf.d/100-glance-api.conf"
 default[:glance][:api][:service_name] = "glance-api"
 
 default[:glance][:registry][:bind_port] = "9191"
 default[:glance][:registry][:log_file] = "/var/log/glance/registry.log"
-default[:glance][:registry][:config_file]="/etc/glance/glance-registry.conf"
+default[:glance][:registry][:config_file] = \
+"/etc/glance/glance-registry.conf.d/100-glance-registry.conf"
```
Comment (Itxaka) [inline chef/cookbooks/glance/attributes/default.rb]: GRRRRRR

Comment (nicolasbock) [inline chef/cookbooks/glance/attributes/default.rb]: Wow, it is touchy :)

File: chef/cookbooks/glance/attributes/default.rb [hunk 3]
```diff
@@ -37,22 +37,22 @@
 default[:glance][:api][:protocol] = "http"
 default[:glance][:api][:bind_port] = "9292"
 default[:glance][:api][:log_file] = "/var/log/glance/api.log"
-default[:glance][:api][:config_file]="/etc/glance/glance-api.conf"
+default[:glance][:api][:config_file]="/etc/glance/glance.conf.d/100-glance-api.conf"
```
Comment (toabctl) [inline chef/cookbooks/glance/attributes/default.rb]: It should be ```/etc/glance/glance-api.conf.d/100-glance-api.conf``` because this config should only be evaluated by the glance-api service

File: chef/cookbooks/glance/attributes/default.rb [hunk 4]
```diff
@@ -37,22 +37,22 @@
...
 default[:glance][:api][:service_name] = "glance-api"
 
 default[:glance][:registry][:bind_port] = "9191"
 default[:glance][:registry][:log_file] = "/var/log/glance/registry.log"
-default[:glance][:registry][:config_file]="/etc/glance/glance-registry.conf"
+default[:glance][:registry][:config_file]="/etc/glance/glance.conf.d/100-glance-registry.conf"
```
Comment (toabctl) [inline chef/cookbooks/glance/attributes/default.rb]: It should be ```/etc/glance/glance-registry.conf.d/100-glance-registry.conf```

File: chef/cookbooks/glance/attributes/default.rb [hunk 5]
```diff
@@ -37,22 +37,22 @@
...
 default[:glance][:registry][:service_name] = "glance-registry"
 
 default[:glance][:cache][:log_file] = "/var/log/glance/cache.log"
-default[:glance][:cache][:config_file]="/etc/glance/glance-cache.conf"
+default[:glance][:cache][:config_file]="/etc/glance/glance.conf.d/100-glance-cache.conf"
```
Comment (toabctl) [inline chef/cookbooks/glance/attributes/default.rb]: It should be ```/etc/glance/glance-cache.conf.d/100-glance-cache.conf```

File: chef/cookbooks/glance/attributes/default.rb [hunk 6]
```diff
@@ -37,22 +37,22 @@
...
 
 default[:glance][:manage][:log_file] = "/var/log/glance/manage.log"
-default[:glance][:manage][:config_file] = "/etc/glance/glance-manage.conf"
+default[:glance][:manage][:config_file] = "/etc/glance/glance.conf.d/100-glance-manage.conf"
```
Comment (toabctl) [inline chef/cookbooks/glance/attributes/default.rb]: It should be ```/etc/glance/glance-manage.conf.d/100-glance-manage.conf```

File: chef/cookbooks/glance/attributes/default.rb [hunk 7]
```diff
@@ -37,22 +37,22 @@
...
 
 default[:glance][:scrubber][:log_file] = "/var/log/glance/scrubber.log"
-default[:glance][:scrubber][:config_file]="/etc/glance/glance-scrubber.conf"
+default[:glance][:scrubber][:config_file]="/etc/glance/glance.conf.d/100-glance-scrubber.conf"
```
Comment (toabctl) [inline chef/cookbooks/glance/attributes/default.rb]: It should be ```/etc/glance/glance-scrubber.conf.d/100-glance-scrubber.conf```

File: chef/cookbooks/glance/recipes/scrubber.rb [hunk 8]
```diff
@@ -44,7 +44,7 @@
     glance_hour: "*",
     glance_user: node[:glance][:user],
     glance_command: "/usr/bin/glance-scrubber "\
-                       "--config-dir #{node[:glance][:config_dir]}")
+                       "--config-file #{node[:glance][:scrubber][:config_file]}")
```
Comment (toabctl) [inline chef/cookbooks/glance/recipes/scrubber.rb]: Can be removed. it is the default when not adding ```--config-file``` switches

File: chef/cookbooks/glance/attributes/default.rb [hunk 9]
```diff
@@ -35,12 +35,12 @@
...
 default[:glance][:api][:bind_port] = "9292"
 default[:glance][:api][:log_file] = "/var/log/glance/api.log"
-default[:glance][:api][:config_file] = "/etc/glance/glance.conf.d/100-glance-api.conf"
+default[:glance][:api][:config_file] = "/etc/glance/glance-api.conf.d/100-glance-api.conf"
 default[:glance][:api][:service_name] = "glance-api"
 
 default[:glance][:registry][:bind_port] = "9191"
 default[:glance][:registry][:log_file] = "/var/log/glance/registry.log"
-default[:glance][:registry][:config_file] = "/etc/glance/glance.conf.d/100-glance-registry.conf"
+default[:glance][:registry][:config_file] = "/etc/glance/glance-registry.conf.d/100-glance-registry.conf"
```
Comment (toabctl) [inline chef/cookbooks/glance/attributes/default.rb]: What about glance-scrubber and glance-manage ?

File: chef/cookbooks/glance/attributes/default.rb (hunk 3 above)
Comment (toabctl) [inline chef/cookbooks/glance/attributes/default.rb]: Already done in a later commit

File: chef/cookbooks/glance/attributes/default.rb (hunk 5 above)
Comment (Itxaka) [inline chef/cookbooks/glance/attributes/default.rb]: Only some of the dirs are created especially for some files. api and registry are both separated because they have colliding options. The rest seem to work ok in the generic config dir as most of their shared options are just repeated (like debug, verbose, registry_port and so on)

Do you think this is needed? This requires a package change to create those dirs. (Can also create them on the recipe but thats not good IMO)

File: chef/cookbooks/glance/recipes/scrubber.rb (hunk 8 above)
Comment (Itxaka) [inline chef/cookbooks/glance/recipes/scrubber.rb]: done

**Comment by Itxaka** [comment]
rebased!
